src/app/components/Providers/LastfmProvider.tsx

```diff
@@ -162,18 +193,69 @@ const LastfmProvider: React.FC<LastfmProviderProps> = ({ children, mk }: LastfmP
     fetchToken();
   }, []);
 
-  useEffect(() => {
-    if (mk.mediaItem && mk.mediaItem.item) {
-      scrobble(mk.mediaItem.item);
-    }
-  }, [mk.mediaItem]);
+  useMK({
+    mediaItem: MusicKit.Events.mediaItemDidChange,
+    playbackState: MusicKit.Events.playbackStateDidChange,
+  });
 
   const state: LastfmProviderValue = {
     login,
     reset,
     connected,
   };
 
+  if (connected) {
+    const track = mk.mediaItem && mk.mediaItem.item;
+    const player = mk.instance.player;
+
+    // Cancels upcoming scrobbles and schedules a new scrobble
+    const scheduleScrobble = () => {
+      clearTimeout(trackStatus.scrobbleTimeoutId);
```

Comment:
Maybe gating `track` here would prevent all the `track && ...` bellow ?

Reply:
I considered this but chose not to because I still want some of the branches below, like [the stopped playing one](https://github.com/Musish/Musish/pull/495/files/392fb99bc8c3f14540514bb253ba2e5d8d863a84#diff-27bc0dbf5dc5cdd33bc33b48dd35e5c5R249-R256), to execute when `track` is not defined.